### ISSUE TITLE
about replicate results from the research papers starting anew

### ISSUE THREAD

**Issue by wl3b10s** [bbc/cnn-fractional-motion-compensation#1]
thanks your source code sharing.

i try to replicate the result based on instructions provided. 

but the performance result i got is much worse than the paper's result.

such as 
for CompetitionCNN RA config, 32frame, -0.35%    (in your paper -0.62% )
for CompetitionCNN RA, BlowingBubbles (D), all frame, -0.4%   (in your paper -1.16%)

and although train epoch is set to 1000, model only trained with only less than 100 epochs then early stopped.

so can you share more detail about your train data generation step?
according to the paper it is : 
trained by BlowingBubbles with QP=27 and RA config.
with Triangle=0, Affine=0, DMVR=0, BIO=0, WeightedPredP=0, WeightedPredB=0
the result dataset is only about 93MB.

is my reproduce method and config  right?
thank you.

**Comment by wl3b10s** [comment]
added  loss log: 
Training COMPETITIONCNN network, from epoch 0
Epoch: [1], time: [16.7617], training loss: [74.80393982], validation loss: [80.61677847]
Epoch: [2], time: [30.2847], training loss: [82.87074280], validation loss: [85.93487676]
Epoch: [3], time: [46.1851], training loss: [55.48914337], validation loss: [60.38148757]
Epoch: [4], time: [61.9862], training loss: [53.75645447], validation loss: [59.50709054]
Epoch: [5], time: [77.7612], training loss: [53.01604843], validation loss: [59.23435691]
Epoch: [6], time: [93.5360], training loss: [49.52455902], validation loss: [58.34114013]
Epoch: [7], time: [109.4552], training loss: [51.93488693], validation loss: [58.71263566]
Epoch: [8], time: [125.2668], training loss: [52.88880539], validation loss: [58.23234484]
Epoch: [9], time: [141.0379], training loss: [51.97753525], validation loss: [57.61796201]
Epoch: [10], time: [156.8787], training loss: [51.25396729], validation loss: [57.63251372]
Epoch: [11], time: [172.6999], training loss: [51.31724930], validation loss: [57.83885279]
Epoch: [12], time: [188.4661], training loss: [50.99987030], validation loss: [57.66149312]
Epoch: [13], time: [204.3262], training loss: [51.40167618], validation loss: [57.84985819]
Epoch: [14], time: [220.1271], training loss: [52.37795258], validation loss: [57.49901224]
Epoch: [15], time: [236.1092], training loss: [51.18250275], validation loss: [57.46840865]
Epoch: [16], time: [251.9359], training loss: [51.29741669], validation loss: [57.14881995]
Epoch: [17], time: [267.7785], training loss: [52.05048370], validation loss: [57.55944627]
Epoch: [18], time: [283.5210], training loss: [52.16976166], validation loss: [57.56901316]
Epoch: [19], time: [299.2875], training loss: [51.48488998], validation loss: [57.23556371]
Epoch: [20], time: [315.0042], training loss: [51.19986343], validation loss: [57.17017192]
Epoch: [21], time: [330.7026], training loss: [50.56499863], validation loss: [57.30763897]
Epoch: [22], time: [346.4610], training loss: [51.70626068], validation loss: [57.50619691]
Epoch: [23], time: [362.3551], training loss: [51.18256760], validation loss: [57.07041513]
Epoch: [24], time: [378.1558], training loss: [51.01570129], validation loss: [57.01261004]
Epoch: [25], time: [393.9694], training loss: [51.14822388], validation loss: [56.97327214]
Epoch: [26], time: [409.8407], training loss: [50.34201050], validation loss: [57.22878376]
Epoch: [27], time: [425.6813], training loss: [50.75917053], validation loss: [56.86641250]
Epoch: [28], time: [441.5453], training loss: [51.20730972], validation loss: [56.99995890]
Epoch: [29], time: [457.3732], training loss: [50.86023331], validation loss: [56.93862915]
Epoch: [30], time: [473.4403], training loss: [50.23540115], validation loss: [56.82676266]
Epoch: [31], time: [489.3776], training loss: [51.18705368], validation loss: [57.12874259]
Epoch: [32], time: [505.3428], training loss: [51.78645706], validation loss: [57.37782817]
Epoch: [33], time: [521.3909], training loss: [50.69324875], validation loss: [56.79302831]
Epoch: [34], time: [537.5163], training loss: [50.93048477], validation loss: [56.93622675]
Epoch: [35], time: [553.5615], training loss: [50.44345856], validation loss: [57.09652968]
Epoch: [36], time: [569.7387], training loss: [51.86124039], validation loss: [56.94158739]
Epoch: [37], time: [585.9923], training loss: [51.52569199], validation loss: [57.16867841]
Epoch: [38], time: [602.4459], training loss: [51.31997681], validation loss: [56.95939464]
Epoch: [39], time: [619.0270], training loss: [50.94802856], validation loss: [56.86762816]
Epoch: [40], time: [635.5671], training loss: [50.78012085], validation loss: [56.86371711]
Epoch: [41], time: [651.8820], training loss: [51.22513962], validation loss: [57.01858262]
Epoch: [42], time: [668.5142], training loss: [50.64779282], validation loss: [56.90445414]
Epoch: [43], time: [685.2439], training loss: [51.26476288], validation loss: [56.64703480]
Epoch: [44], time: [702.0090], training loss: [51.19293976], validation loss: [57.27315374]
Epoch: [45], time: [718.6485], training loss: [50.61373520], validation loss: [56.82956732]
Epoch: [46], time: [735.3232], training loss: [50.78625870], validation loss: [56.98439026]
Epoch: [47], time: [751.9847], training loss: [50.45402527], validation loss: [56.85384750]
Epoch: [48], time: [768.6154], training loss: [50.31046295], validation loss: [56.92136592]
Epoch: [49], time: [785.1841], training loss: [51.08269882], validation loss: [57.10692387]
Epoch: [50], time: [801.8229], training loss: [50.25175858], validation loss: [56.95656795]
Epoch: [51], time: [818.3226], training loss: [50.53646851], validation loss: [56.91735643]
Epoch: [52], time: [835.0490], training loss: [50.98997498], validation loss: [56.90726520]
Epoch: [53], time: [851.6359], training loss: [50.71559143], validation loss: [56.78072013]
Epoch: [54], time: [868.1443], training loss: [50.97849274], validation loss: [56.71111556]
Epoch: [55], time: [884.7035], training loss: [50.37860870], validation loss: [56.92308438]
Epoch: [56], time: [901.1966], training loss: [50.78044510], validation loss: [56.69400578]
Epoch: [57], time: [917.8126], training loss: [51.45368195], validation loss: [56.85925896]
Epoch: [58], time: [934.2990], training loss: [51.50955582], validation loss: [56.78827556]
Epoch: [59], time: [950.7882], training loss: [51.29473495], validation loss: [57.06541640]
Epoch: [60], time: [967.4304], training loss: [51.10828400], validation loss: [56.96873240]
Epoch: [61], time: [983.8231], training loss: [50.87610626], validation loss: [56.84607401]
Epoch: [62], time: [1000.6121], training loss: [50.93948746], validation loss: [56.74977481]
Epoch: [63], time: [1017.5928], training loss: [50.17265701], validation loss: [56.91001363]
Epoch: [64], time: [1034.3110], training loss: [50.97959900], validation loss: [56.60984642]
Epoch: [65], time: [1050.9997], training loss: [51.15404892], validation loss: [56.84727220]
Epoch: [66], time: [1067.7521], training loss: [50.52456665], validation loss: [56.62690415]
Epoch: [67], time: [1084.5992], training loss: [51.36140442], validation loss: [56.95543228]
Epoch: [68], time: [1101.4819], training loss: [51.34338379], validation loss: [56.87696715]
Epoch: [69], time: [1118.6422], training loss: [50.87261200], validation loss: [56.82119837]
Epoch: [70], time: [1135.6697], training loss: [51.62710953], validation loss: [56.79853402]
Epoch: [71], time: [1152.6066], training loss: [50.96104431], validation loss: [56.67465505]
Epoch: [72], time: [1169.7924], training loss: [51.12660217], validation loss: [56.71354318]
Epoch: [73], time: [1186.6895], training loss: [51.62648010], validation loss: [56.69665970]
Epoch: [74], time: [1203.5547], training loss: [50.60503387], validation loss: [56.68591542]
Epoch: [75], time: [1220.4102], training loss: [51.02727890], validation loss: [56.85564213]
Epoch: [76], time: [1237.1940], training loss: [51.31077576], validation loss: [56.70337086]
Epoch: [77], time: [1254.0658], training loss: [51.39686584], validation loss: [56.81269000]
Epoch: [78], time: [1270.7702], training loss: [50.51595306], validation loss: [56.62065161]
Epoch: [79], time: [1287.5176], training loss: [50.99675751], validation loss: [56.70802418]
Epoch: [80], time: [1304.2347], training loss: [51.45772171], validation loss: [56.77631772]
Epoch: [81], time: [1320.9238], training loss: [51.05890656], validation loss: [56.77573542]
Epoch: [82], time: [1337.7547], training loss: [51.27669907], validation loss: [56.95365610]
Epoch: [83], time: [1354.8236], training loss: [50.75201035], validation loss: [56.86759506]
Epoch: [84], time: [1371.7571], training loss: [51.33850479], validation loss: [56.78693907]
Epoch: [85], time: [1388.6998], training loss: [51.39620209], validation loss: [56.71142147]
Epoch: [86], time: [1405.4855], training loss: [50.96632385], validation loss: [56.73093623]
Epoch: [87], time: [1422.5008], training loss: [50.96469879], validation loss: [56.72883089]
Epoch: [88], time: [1439.3723], training loss: [50.99902344], validation loss: [56.73554353]
Epoch: [89], time: [1456.3717], training loss: [51.56198883], validation loss: [56.68959045]
Epoch: [90], time: [1473.2350], training loss: [51.32006073], validation loss: [56.62522790]
Epoch: [91], time: [1490.2688], training loss: [51.44070816], validation loss: [57.25827026]
Epoch: [92], time: [1507.3484], training loss: [51.73938751], validation loss: [56.93950235]
Epoch: [93], time: [1524.2520], training loss: [51.09768295], validation loss: [56.88435647]
Epoch: [94], time: [1541.1066], training loss: [50.86421967], validation loss: [56.79212090]
Epoch: [95], time: [1557.9325], training loss: [50.85690689], validation loss: [57.09913512]
Epoch: [96], time: [1574.7921], training loss: [50.98610687], validation loss: [56.76048980]
Epoch: [97], time: [1591.7463], training loss: [51.21313095], validation loss: [57.32886850]
Epoch: [98], time: [1608.5849], training loss: [51.40014648], validation loss: [56.58277684]
Epoch: [99], time: [1625.4559], training loss: [51.17512512], validation loss: [56.67546475]
Epoch: [100], time: [1642.3969], training loss: [51.34023666], validation loss: [56.76897123]
Epoch: [101], time: [1659.2905], training loss: [50.83700943], validation loss: [56.80917321]
Epoch: [102], time: [1676.1677], training loss: [51.00191116], validation loss: [56.63593058]
Epoch: [103], time: [1693.1028], training loss: [50.99552917], validation loss: [56.80756157]
Epoch: [104], time: [1710.0468], training loss: [51.27416992], validation loss: [56.67572612]
Epoch: [105], time: [1726.9893], training loss: [51.00672150], validation loss: [56.60385895]
Epoch: [106], time: [1743.8022], training loss: [51.56180573], validation loss: [56.54008927]
Epoch: [107], time: [1760.7787], training loss: [51.34359360], validation loss: [56.63075675]
Epoch: [108], time: [1777.8932], training loss: [50.84634018], validation loss: [56.65110533]
Epoch: [109], time: [1794.8159], training loss: [51.24534225], validation loss: [56.72056050]
Epoch: [110], time: [1811.7397], training loss: [51.54280090], validation loss: [56.53592977]
Epoch: [111], time: [1828.6706], training loss: [51.48693085], validation loss: [56.62228098]
Epoch: [112], time: [1845.5385], training loss: [51.23863983], validation loss: [57.05349153]
Epoch: [113], time: [1862.4018], training loss: [51.27053070], validation loss: [56.73976800]
Epoch: [114], time: [1879.1688], training loss: [51.36951447], validation loss: [56.80685708]
Epoch: [115], time: [1895.8579], training loss: [51.74368668], validation loss: [56.69389959]
Epoch: [116], time: [1912.7525], training loss: [51.55915070], validation loss: [56.61990480]
Epoch: [117], time: [1929.2979], training loss: [51.36177063], validation loss: [56.66944971]
Epoch: [118], time: [1946.0271], training loss: [51.05568695], validation loss: [56.82140018]
Epoch: [119], time: [1962.7148], training loss: [51.72304535], validation loss: [56.67713153]
Epoch: [120], time: [1979.3588], training loss: [51.86597443], validation loss: [56.73485491]
Epoch: [121], time: [1995.9950], training loss: [51.40619659], validation loss: [56.58572781]
Epoch: [122], time: [2012.8946], training loss: [51.57538986], validation loss: [56.65356913]
Epoch: [123], time: [2029.6674], training loss: [51.58467865], validation loss: [56.75821563]
Epoch: [124], time: [2046.4981], training loss: [51.52143478], validation loss: [56.72449370]
Epoch: [125], time: [2063.4506], training loss: [51.34130478], validation loss: [56.63582020]
Epoch: [126], time: [2080.1790], training loss: [51.76092148], validation loss: [56.56482770]
Epoch: [127], time: [2096.9769], training loss: [51.21775055], validation loss: [56.77636522]
Epoch: [128], time: [2113.4514], training loss: [51.15604401], validation loss: [56.68215573]
Epoch: [129], time: [2129.7427], training loss: [51.28002167], validation loss: [56.66595939]
Epoch: [130], time: [2146.0949], training loss: [51.60972595], validation loss: [56.73169794]
Epoch: [131], time: [2162.3787], training loss: [51.02887344], validation loss: [56.66092227]
Epoch: [132], time: [2178.7411], training loss: [51.20776749], validation loss: [56.50389173]
Epoch: [133], time: [2195.0203], training loss: [51.17226791], validation loss: [56.77413239]
Epoch: [134], time: [2211.4294], training loss: [50.83421707], validation loss: [56.63581824]
Epoch: [135], time: [2227.7030], training loss: [51.56256866], validation loss: [56.73935232]
Epoch: [136], time: [2244.2004], training loss: [51.09438705], validation loss: [56.71956068]
Epoch: [137], time: [2260.6892], training loss: [50.60120392], validation loss: [56.80351725]
Epoch: [138], time: [2277.1382], training loss: [51.25839615], validation loss: [56.57732305]
Epoch: [139], time: [2293.4701], training loss: [51.50028610], validation loss: [56.65371630]
Epoch: [140], time: [2309.9876], training loss: [51.62242508], validation loss: [56.70872165]
Epoch: [141], time: [2326.3050], training loss: [51.54183578], validation loss: [56.74675172]
Epoch: [142], time: [2342.9024], training loss: [51.29401016], validation loss: [56.81945653]
Epoch: [143], time: [2359.3819], training loss: [50.85771942], validation loss: [56.88126066]
Epoch: [144], time: [2376.0233], training loss: [51.36215973], validation loss: [56.70672263]
Epoch: [145], time: [2392.6929], training loss: [51.76704407], validation loss: [56.69957930]
Epoch: [146], time: [2409.5777], training loss: [50.95315933], validation loss: [56.84875353]
Epoch: [147], time: [2426.2395], training loss: [51.14400101], validation loss: [56.98768960]
Epoch: [148], time: [2442.9195], training loss: [51.84106445], validation loss: [56.65256599]
Epoch: [149], time: [2459.7385], training loss: [51.41394806], validation loss: [56.64097029]
Epoch: [150], time: [2476.8577], training loss: [51.23407364], validation loss: [56.85905457]
Epoch: [151], time: [2493.7567], training loss: [51.32021332], validation loss: [56.67542168]
Epoch: [152], time: [2510.6367], training loss: [51.10459137], validation loss: [56.66326104]
Epoch: [153], time: [2527.5625], training loss: [50.82736206], validation loss: [56.71049832]
Epoch: [154], time: [2544.4687], training loss: [51.14965439], validation loss: [56.73730616]
Epoch: [155], time: [2561.3310], training loss: [51.03409195], validation loss: [56.80878030]
Epoch: [156], time: [2578.2687], training loss: [50.94479370], validation loss: [56.62634376]
Epoch: [157], time: [2595.0220], training loss: [50.97457886], validation loss: [56.81865077]
Epoch: [158], time: [2611.8578], training loss: [51.68045425], validation loss: [56.80046980]
Epoch: [159], time: [2628.5457], training loss: [51.11251450], validation loss: [56.65910290]
Epoch: [160], time: [2645.2744], training loss: [51.59649277], validation loss: [56.66719277]
Epoch: [161], time: [2661.9163], training loss: [51.73128510], validation loss: [56.62581967]
Epoch: [162], time: [2678.7680], training loss: [51.19163895], validation loss: [56.65550404]
Epoch: [163], time: [2695.8882], training loss: [50.60602570], validation loss: [56.65047221]
Epoch: [164], time: [2712.6286], training loss: [51.52814484], validation loss: [56.66649259]
Epoch: [165], time: [2729.5093], training loss: [51.50249481], validation loss: [56.56410267]
Epoch: [166], time: [2746.4827], training loss: [51.04123306], validation loss: [56.89800349]
Epoch: [167], time: [2763.4092], training loss: [50.67354965], validation loss: [56.86413365]
Epoch: [168], time: [2780.1433], training loss: [51.50175858], validation loss: [56.55916251]
Epoch: [169], time: [2796.9254], training loss: [51.70675659], validation loss: [56.63857786]
Epoch: [170], time: [2813.8840], training loss: [51.29373550], validation loss: [56.72264148]
Epoch: [171], time: [2830.8042], training loss: [50.89037323], validation loss: [56.82236998]
Epoch: [172], time: [2847.8920], training loss: [51.02082443], validation loss: [56.85424952]
Epoch: [173], time: [2864.8178], training loss: [51.00419235], validation loss: [56.70261691]
Epoch: [174], time: [2881.7956], training loss: [50.59973145], validation loss: [56.79454545]
Epoch: [175], time: [2898.8280], training loss: [50.63301086], validation loss: [57.02535949]
Epoch: [176], time: [2916.0431], training loss: [50.84169006], validation loss: [56.86154076]
Epoch: [177], time: [2933.0541], training loss: [51.40530396], validation loss: [56.50748493]
Epoch: [178], time: [2950.1689], training loss: [51.24463272], validation loss: [56.67069884]
Epoch: [179], time: [2967.2894], training loss: [51.31899643], validation loss: [56.62312071]
Epoch: [180], time: [2984.1058], training loss: [50.78177261], validation loss: [56.78749380]
Epoch: [181], time: [3000.9948], training loss: [51.37445831], validation loss: [56.70139879]

**Comment by lmurn** [comment]
Hello, 

New commits have been added that adress and fix the specified issue. The problem occured due to incorrect conditions set in the VTM6.0 patch, a bug in the YUVReader Python class and erroneous CompetitionCNN training process. 

Apologies for the inconvenience, the v1.1 commit should produce reliable and accurate results. Thank you for bringing these problems to our attention!
